spring-cloud-aws-sns/src/main/java/io/awspring/cloud/sns/sms/attributes/SmsMessageAttributes.java

```diff
@@ -0,0 +1,338 @@
+/*
+ * Copyright 2013-2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.awspring.cloud.sns.sms.attributes;
+
+import static io.awspring.cloud.sns.sms.attributes.ConvertToMessageAttributes.populateMapWithNumberValue;
+import static io.awspring.cloud.sns.sms.attributes.ConvertToMessageAttributes.populateMapWithStringValue;
+
+import java.util.HashMap;
+import java.util.Map;
+import software.amazon.awssdk.services.sns.model.MessageAttributeValue;
+
+/**
+ * @author Matej Nedic
+ * @since 3.0.0
+ */
+public class SmsMessageAttributes {
+	private String senderID;
+	private String originationNumber;
+	private String maxPrice;
+	private SmsType smsType;
+	private String entityId;
+	private String templateId;
+	private String messageGroupId;
+	private String deduplicationId;
+	private String messageStructure;
```

Comment:
As far as I can see, the only valid value for message structure is "json" and it is used for sending to topic when more than single protocol endpoints are subscribed to this topic. I think this is "advanced" usage that we don't need to cover with `SnsSmsTemaplte` - we can support sending only to topics with sms endpoints or if we decide to support mixed topics, this has to be redesigned.